improve ionization documentation (#4944)
ComputationalRadiationPhysics/picongpu: docs/source/usage/workflows/ionization.rst

                {
                    static constexpr float_X numberOfProtons  = 7.0;
                    static constexpr float_X numberOfNeutrons = 7.0;
                };
            } // namespace picongpu::ionization::atomicNumbers

    2.) effective nuclear charge for each charge state

        .. code-block:: c++

            /* Example: nitrogen */
            namespace picongpu::ionization::effectiveNuclearCharge
            {
                /// @attention insert the values in REVERSE order since the lowest shell corresponds to the last ionization process.
                PMACC_CONST_VECTOR(
                    float_X,
                    7, // number of charge states
                    Nitrogen, // name you may reference this by later, remember to prepend the namespace!
                    /* 2p^3 */
                    3.834,
                    3.834,
                    3.834,
                    /* 2s^2 */
                    3.874,
                    3.874,
                    /* 1s^2 */
                    6.665,
                    6.665);
            } // namespace picongpu::ionization::effectiveNuclearCharge

        .. note::

>             see `wikipedia <https://en.wikipedia.org/wiki/Effective_nuclear_charge>`_ for values or refer directly to the calculations by Slater or Clementi and Raimondi, [1]_ and [2]_.

Reviewer: Neither reference 1 nor 2 is a "Slater" paper. Both are Clementi and Raimondi.

Author: this was copied from the FoilLTC example, removed the reference to Slater,

3. Define mass and charge ratios of ions in the :code:`speciesDefinition.param`.

    .. code-block:: c++

        /* Example Nitrogen*/
        namespace picongpu
        {
            /* mass and charge ratios with respect to electrons */
            value_identifier(float_X, MassRatioNitrogen, 25514.325);
            value_identifier(float_X, ChargeRatioNitrogen, -7.0);
        } // namespace picongpu

4. Define at least one electron species in :code:`speciesDefinition.param`.

    .. code-block:: c++

        /* Example electron Species */
        namespace picongpu
        {
            using ParticleFlagsElectrons = MakeSeq_t<
                particlePusher<UsedParticlePusher>,
                shape<UsedParticleShape>,
                interpolation<UsedField2Particle>,
                current<UsedParticleCurrentSolver>,
                massRatio<MassRatioElectrons>,
                chargeRatio<ChargeRatioElectrons>>;

            using ParticleAttributesElectrons = MakeSeq_t<position<position_pic>, momentum, weighting>;

            using Electrons = Particles<PMACC_CSTRING("e"), ParticleFlagsElectrons, ParticleAttributesElectrons>;
        } // namespace picongpu

> 5. Define for each ion species a PIConGPU species

Reviewer: This sentence makes no sense. What is a "PIConGPU species"?

Author: done

>     The definition of an ioniaztion species must include the particle attribute :code:`boundElectrons`, and the particle flags :code:`atomicNumbers`, :code:`ionizationEnergies`, :code:`effectiveNuclearCharge`, :code:`ionizers`, :code:`massRatio` and :code:`chargeRatio`.

Reviewer: typo
```suggestion
    The definition of an ionization species must include the particle attribute :code:`boundElectrons`, and the particle flags :code:`atomicNumbers`, :code:`ionizationEnergies`, :code:`effectiveNuclearCharge`, :code:`ionizers`, :code:`massRatio` and :code:`chargeRatio`.
```

Author: done

                ionizationEnergies<ionization::energies::AU::Nitrogen_t >               // <-- from step 2
                massRatio<MassRatioNitrogen>,                                           // <-- from step 3
                chargeRatio<ChargeRatioNitrogen>,                                       // <-- from step 3
                ionizers<MakeSeq_t<               // <-- comma separated list of all ionizers of this species with
                                                  // species of electrons macro particles to be created upon ionization
                    particles::ionization::BSIEffectiveZ<Electrons, particles::ionization::current::PARAM_IONIZATIONCURRENT>,
                    particles::ionization::ADKLinPol<Electrons, particles::ionization::current::PARAM_IONIZATIONCURRENT>,
                    particles::ionization::ThomasFermi<Electrons>>>>;

            using ParticleAttributesNitrogen = MakeSeq_t<position<position_pic>, momentum, weighting, boundElectrons>;
            using Nitrogen = Particles<PMACC_CSTRING("N"), ParticleFlagsNitrogen, ParticleAttributesNitrogen>;
        } // namespace picongpu

    .. note::

>         Remember that you can define multiple electron species and associate them with your different ion species to track the ionisation processes separately!

Reviewer: change from AE to BE - please stick to AE
```suggestion
        Remember that you can define multiple electron species and associate them with your different ion species to track the ionization processes separately!
```

Author: done

6. Create ion macro particles in the simulation and initialize them in :code:`speciesInitialization.param`.

    .. code-block:: c++

        namespace picongpu::particles
        {
            /* create nitrogen charge state 1 and electrons to neutralize the simulation */
            using InitPipeline = pmacc::mp_list<
                CreateDensity< ... some densityProfile... , startPosition::Random, Nitrogen>,
                ManipulateDerive<manipulators::binary::DensityWeighting, Nitrogen, Electrons>,
                Manipulate<manipulators::unary::ChargeState<1u>;, Nitrogen>>;
        } // namespace picongpu::particles

    .. attention::

>         Remember to initialize the simulation charge neutral to avoid creating ghost background charge densities.

Reviewer: This is not really a problem. For many simulations, you work with the ion "ghost" background. How is the following sentence:
```suggestion
        Ensure the simulation is initialized as charge neutral whenever the respective species should move, to avoid creating unintended ghost background charge densities.
```

Author: done

.. note::
    For an example of a picongpu setup with ionization see the `FoilLTC example <https://github.com/ComputationalRadiationPhysics/picongpu/tree/dev/share/picongpu/examples/FoilLCT>`_.

.. [1]
    Clementi, E.; Raimondi, D. L. (1963)
    "Atomic Screening Constants from SCF Functions"
    J. Chem. Phys. 38 (11): 2686-2689.
    https://dx.doi.org/10.1063/1.1733573
.. [2]
    Clementi, E.; Raimondi, D. L.; Reinhardt, W. P. (1967)
    "Atomic Screening Constants from SCF Functions. II. Atoms with 37 to 86 Electrons"
    Journal of Chemical Physics. 47: 1300-1307
    https://dx.doi.org/10.1063/1.1712084
.. [3]
    Kramida, A., Ralchenko, Yu., Reader, J., and NIST ASD Team (2014)
    NIST Atomic Spectra Database (ver. 5.2), [Online]

>     Available: http://physics.nist.gov/asd [2017, February 8]

Reviewer: Optional, perhaps use the DOI instead of the direct URL:
```suggestion
    Available: https://dx.doi.org/10.18434/T4W30F [2017, February 8]
```

Author: done
